samples/sample-sync-doc-definitions.js

```diff
@@ -308,6 +308,7 @@ function() {
       typeFilter: function(doc, oldDoc) {
         return createBusinessEntityRegex('notification\\.[A-Za-z0-9_-]+\\.processedTransport\\.[A-Za-z0-9_-]+$').test(doc._id);
       },
+      cannotDelete: true,
```

Comment:
A good example for this constraint. The document's properties are largely immutable except for its `sentAt` property, which will be set some time after it is created, so the document itself cannot be marked as `immutable`, but it also should never be deleted.